db/view/view.cc

```diff
@@ -1762,42 +1809,41 @@ get_view_natural_endpoint(
             // otherwise.
             if (it != base_endpoints.end()) {
                 base_endpoints.erase(it);
-            } else if (!network_topology || view_topology.get_datacenter(view_endpoint) == my_datacenter) {
-                view_endpoints.push_back(view_endpoint);
-            }
-        } else {
-            if (!network_topology || view_topology.get_datacenter(view_endpoint) == my_datacenter) {
-                view_endpoints.push_back(view_endpoint);
+            } else {
+                auto np = resolve(view_topology, view_endpoint, true);
+                if (!network_topology || (np && np->dc() == my_datacenter)) {
+                    view_endpoints.push_back(*np);
+                }
             }
         }
+    } else {
+        for (auto&& view_endpoint : view_erm->get_replicas(view_token)) {
+            process_candidate(view_endpoints, view_topology, view_endpoint, true);
+        }
+    }
+
+    if (rack_aware_pairing && !simple_rack_aware_pairing) {
+        std::ranges::stable_sort(base_endpoints, std::ranges::less{}, std::mem_fn(&locator::node::rack));
```

Comment:
shouldn't we ditch simple_rack_aware_pairing?

it can happen that replica lists aren't sorted by rack in base and view, even when RF == #_of_racks. the tablet allocator is greedy, and it can happen replicas aren't allocated in rack order.

simple_rack_aware_pairing sounds like a optimization for bypassing sorting, right?

Reply:
Right, it skips sorting and also filters by matching location (dc and rack), so it selects only the replicas in the node's rack in both base and view.